### DISCUSSION TITLE
CLR Path / .net exe in a different directory from Rust exe

### DISCUSSION THREAD

**Discussion by RandallOfLegend** [yamakadi/clroxide#10] [Q&A]
I have a .net executable that references other DLL's. If I place the .net executable in a separate directoryfrom the Rust executable then my .net executable is unable to find the DLLs.  If you print the result of "Assembly.GetExecutingAssembly().Location" (C#) when running from Visual Studio you will get the correct location, but from Rust/ClrOxide, you will get an empty string. This makes sense given how ClrOxide is reading the bytes from the exe into the CLR contents before running.

Is there a workaround where I can have all of my .net code in one directory, and the Rust code in another? Currently I have to run everything out of the same directory so my .net code can locate the DLLs.

**Comment by yamakadi** [comment]
Apologies for the very late response.

1) You can try to put everything into a single DLL using FodyWeavers
2) You can forego the existing execute assembly method, which uses the _AppDomain interface, and use the “Assembly” type and its methods directly. There’s an example of using `LoadWithPartialName` is the new powershell example.
